cps/kni.c

```diff
@@ -44,6 +44,61 @@
 extern long init_module(void *, unsigned long, const char *);
 extern long delete_module(const char *, unsigned int);
 
+/*
+ * The minimum length of each update is a Netlink message header,
+ * a routing table message header, an IPv4 address for the route,
+ * an IPv4 address for the gateway, and a 32-bit interface ID.
+ *
+ */
+#define MIN_RTNL_UPDATE_LEN (sizeof(struct nlmsghdr) + sizeof(struct rtmsg) + \
+	2 * sizeof(struct in_addr) + sizeof(uint32_t))
+
+/* Maximum number of route updates that can fit in a Netlink message. */
+#define MAX_RTNL_UPDATES (MNL_SOCKET_BUFFER_SIZE / MIN_RTNL_UPDATE_LEN)
+
+struct route_update {
+	/* Type of route update: RTM_NEWROUTE or RTM_DELROUTE. */
+	int      type;
+
+	/* Address family of route: AF_INET or AF_INET6. */
+	int      family;
+
+	/*
+	 * Whether this update has all the fields and attributes
+	 * necessary to update the LPM table.
+	 */
+	int      valid;
+
+	/* Route prefix length. */
+	uint8_t  prefix_len;
+
+	/* Output interface index of route. */
+	uint32_t oif_index;
+
+	/* Route origin. See field rtm_protocol of struct rtmsg. */
+	uint8_t  rt_proto;
```

Comment:
Moving field `rt_proto` just after field `prefix_len` will reduce the size of `struct route_update` due to the alignment that the compiler calculate.